addOns/ascanrulesBeta/src/main/java/org/zaproxy/zap/extension/ascanrulesBeta/XxeScanRule.java

```diff
@@ -407,4 +456,37 @@ static String createLfrPayload(String requestBody) {
         sb.append(requestBody.substring(endIdx));
         return sb.toString();
     }
+
+    private int getTotalTagCount(String requestBody) {
+
+        // Remove original xml header
+        Matcher headerMatcher = xmlHeaderPattern.matcher(requestBody);
```

Comment:
This should not be duplicated.

Reply:
@thc202  I've made some changes, but am still unsure of exactly what the tests lack. Do I need to check for another payload like 

`String localFileInclusionAttackPayload =
                MessageFormat.format(XxeScanRule.ATTACK_HEADER, "file:///etc/passwd")
                        + "<comment><text>test</text></comment>\n"
                        + "<text>&zapxxe;</text>";`